codalab/lib/bundle_util.py

```diff
@@ -354,3 +354,63 @@ def check_bundle_freezable(bundle):
             'Cannot freeze bundle %s(%s), bundle is not in a final state.'
             % (bundle.uuid, bundle.metadata.to_dict()["name"])
         )
+
+
+def get_bundle_state_details(bundle):
+    """
+    This helper returns information about what the given bundle's state means.
+    State definitions varry slightly based on the bundle's type.
+
+    If the bundle is currently running, its run status will be returned.
+    If the bundle is currently staged, its staged status will be returned.
+
+    Otherwise, a description of the state is returned.
+    """
+    if 'state' not in bundle or 'bundle_type' not in bundle:
```

Comment:
Does this happen?

Reply:
During testing I didn't observe this, but I wanted to add this as a safety check juuuuust in case.

Reply:
Hm, ideally we shouldn't add these sort of defensive checks since it could mask subtle problems...  One intermediate solution is to log these errors so we fail gracefully but we can track when things don't behave as expected.